classes/class-ppc-loader.php

```diff
@@ -47,33 +47,46 @@ public static function get_instance() {
 		public function __construct() {
 			$this->ppc_default_list_data();
 			$this->ppc_load();
+			$this->ppc_update();
 			add_action( 'admin_enqueue_scripts', array( $this, 'ppc_plugin_backend_js' ) );
 			add_action( 'init', array( $this, 'ppc_save_data' ) );
 			add_action( 'wp_ajax_ppc_checklistitem_add', array( $this, 'ppc_add_item' ), 1 );
 			add_action( 'wp_ajax_ppc_checklistitem_delete', array( $this, 'ppc_delete_item' ), 1 );
 			add_action( 'wp_ajax_ppc_checklistitem_drag', array( $this, 'ppc_drag_item' ), 1 );
 			add_action( 'wp_ajax_ppc_checklistitem_edit', array( $this, 'ppc_edit_item' ), 1 );
 		}
+
+		/**
+		 * Check for default list if present with a default parameter.
+		 *
+		 * @param string $post_type Post type.
+		 */
+		public function get_list_by_post_type( $post_type = '' ) {
+			$ppc_cpt_checklist       = self::get_instance()->get_list();
+			$ppc_checklist_item_data = array();
+			if ( ! empty( $ppc_cpt_checklist ) && ! empty( $post_type ) && isset( $ppc_cpt_checklist[ $post_type ] ) ) {
+				$ppc_checklist_item_data = $ppc_cpt_checklist[ $post_type ];
+			}
+			return $ppc_checklist_item_data;
+		}
+
+		/**
+		 * Stores default checklist in the database.
+		 */
+		public function get_list() {
+			return get_option( 'ppc_cpt_checklist_data', array() );
+		}
+
 		/**
 		 * Stores default list in the database.
 		 *
 		 * @since 1.0
 		 * @return void
 		 */
 		public function ppc_default_list_data() {
-			$ppc_default_checklist_data = array(
-				'ppc_key2' => 'Featured Image Assigned',
-				'ppc_key3' => 'Category Selected',
-				'ppc_key4' => 'Formatting Done',
-				'ppc_key5' => 'Title is Catchy',
-				'ppc_key6' => 'Social Images Assigned',
-				'ppc_key7' => 'Done SEO',
-				'ppc_key8' => 'Spelling and Grammar Checked',
-			);
-			$ppc_default_post_types     = array( 'post', 'page' );
-			add_option( 'ppc_checklist_data', $ppc_default_checklist_data );
-			add_option( 'ppc_post_types_to_display', $ppc_default_post_types );
+			add_option( 'ppc_post_types_to_display', array( 'post', 'page' ) );
```

Comment:
This add_option will load on every load. If this is a static array. Declare this as a static variable.